### PR TITLE
Week-1 task Completed

### PR DESCRIPTION
**Pull request by shailesh1028**
### Week-1 Task
**Layout testing**: Created test to check the different version/layout of preview based on the API data. This will help when we run the tests on different viewports.
Apart from this, added an event test that includes hovering and clicking over the span to generate the preview.
Error Found: The close button does not work in all other modes except in Standard mode.

### CODE REVIEW

**Comment by jpita** [comment]
1. run `npx eslint cypress --fix` 
2. install the needed dependencies
3. put this on the `.eslintrc.js` file
```
module.exports = {
  env: {
    browser: true,
    es6: true
  },
  extends: [
    'eslint:recommended',
    'wikimedia/client',
    "plugin:cypress/recommended"
  ],
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
    require: true
  },
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module'
  },
  rules: {
    semi: [2, "never"],
    'no-nonoctal-decimal-escape': 'off',
    "no-only-tests/no-only-tests": "error",
    "cypress/no-unnecessary-waiting": "warn"
  },
  "plugins": [
    "cypress",
    "no-only-tests"
  ],
}

```

4. run `npx eslint cypress --fix` 
5. fix the issues

File: cypress/integration/event/active.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { Preview } from '../../page-objects/preview'
+
+const preview = new Preview()
+
+
+describe('Check the Hover and Click Event', () => {
+    
+    beforeEach('Open the English Page', () => {
+        cy.navigateToHomePage('/articles/english.html')
+        cy.wait(2000)
```
[the remark this answers is not in the document]
Comment (shailesh1028) [reply cypress/integration/event/active.js]: The wait is required in my system as it takes time to load the changes. I have removed it while pushing the current changes.

File: cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    preview.Header().should('be.visible')
+    preview.HeaderWatermark().should('be.visible')
+    preview.HeaderClosebtn().should('be.visible')
+
+    preview.Preview().then( tag => {
+
+        const lang = tag.attr('lang')
+
+        preview.Body().should('be.visible')
+        .and('have.class','wikipediapreview-body-disambiguation')
```
[the remark this answers is not in the document]
Comment (shailesh1028) [reply cypress/support/commands.js]: I do understand your concern. But those classes are added during runtime to render the different layouts. Thereby it is necessary to check whether they are added or not.

File: cypress/support/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import './commands'
+
+
+
+Cypress.on('uncaught:exception', (err, runnable) => {
```
[the remark this answers is not in the document]
Comment (shailesh1028) [reply cypress/support/index.js]: It is required when the preview reacts unexpectedly due to some issues(mainly connectivity issues as it delays the response ) so this command prevents the cypress test to fail due to the errors generated from our app.

File: cypress/integration/event/active.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/event/active.js]: for slow networks u can use 
https://docs.cypress.io/guides/references/configuration#Timeouts
in the configuration file `cypress.json` to increase the timeout

File: cypress/support/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/support/index.js]: a test should always fail whenever something goes wrong, this shouldn't be used

File: cypress/support/commands.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/support/commands.js]: so don't use commands.js for this, use page-objects elements that are only used searched for when needed.
avoid using commands.js for specific test logic, only when it's for actions on the app that always use present elements like we do here - https://github.com/wikimedia/wikipedia-kaios/blob/main/cypress/support/commands.js

u can delete most of ur commands.js code and move it to test and page-object files

**Comment by jpita** [comment]
did u do steps 4 and 5 from my previous comment?

File: cypress/page-objects/preview.js [hunk 4]
```diff
@@ -1,79 +1,80 @@
...
 
+	getBodyAction() {
+		return cy.get( 'div.wikipediapreview' ).find( 'div.wikipediapreview-body-action > a' )
+	}
 
+	//  Wikipedia Preview Footer
 
-    //  Wikipedia Preview Footer
+	getFooter() {
+		return cy.get( 'div.wikipediapreview' ).find( ' div.wikipediapreview-footer' )
```
Comment (jpita) [inline cypress/page-objects/preview.js:69]: why not do 
`cy.get('div.wikipediapreview-footer')` 

only instead of get+find?

**Comment by jpita** [comment]
you don't need the check.js class, delete it and add the methods to check for elements inside the preview page object.
check what we did in the kaios code https://github.com/wikimedia/wikipedia-kaios/tree/main/cypress/page-objects.

when u have a page object class, you can create a method inside that page object to check for the presence of the elements.
the same way u would have a login method inside the login page object.
the purpose is to keep the code organised and easy to change when there's a selector change or a UI change.
u can read up on page-object 
https://www.guru99.com/page-object-model-pom-page-factory-in-selenium-ultimate-guide.html
https://www.browserstack.com/guide/page-object-model-in-selenium


if u don't understand what I mean we can talk about this during a meeting.

P.S.: similarly to the kaios files, ur page objects need to end in `page.js`. If u have some files that are not page-objects please remove them from the folder and create a different one, for example `helpers`

**Pull request merged**